software/ADPL_electron/ADPL_electron.ino

```diff
@@ -200,9 +213,14 @@ int publish_data(int last_publish_time) {
             tempHXCI.temp, tempHXCO.temp, tempHTR.temp, tempHXHI.temp, tempHXHO.temp,
             int(valve.gasOn), int(bucket.tip_count));
 
-    publish_success = Particle.publish("DATA",data_str);
+    if (sdFile) {
```

Comment:
hmm not sure how this evaluates if `sdFile` is never declared for a file with no SD card? Due to the `#if SDCARD` block? You probably know better than I do here.